### ISSUE TITLE
Underscoreis treated as a part of reference

### ISSUE THREAD

**Issue by grauwoelfchen** [flying-sheep/rust-rst#14]
*UPDATED*:

This seems to happen also in `paragraph` or `emphasis` etc. So, I'll update the title of this issue.

----

Hi,

I've just found something a little bit strange parse result of `literal` (inline code), in case, that includes underscore(s) like `NAME_WITH_UNDERSCORE`.

I confirmed that by adding test cases in `parser/src/tests.rs` as followings.

Without underscore, It worked as expected:

```rust
#[test]
fn inline_code_literal_with_underscore() {
    parses_to! {
        parser: RstParser,
        input: "``NAME``",
        rule: Rule::inline,
        tokens: [
            literal(2, 6, [
                str_nested(2, 6),
            ]),
        ]
    };
    ...
}
```

But with underscore(s), I got the result like this:

```rust
#[test]
fn inline_code_literal_with_underscore() {
    ...
    parses_to! {
        parser: RstParser,
        input: "``NAME_WITH_UNDERSCORE``",
        rule: Rule::inline,
        tokens: [
            // I expected this...
            // literal(2, 22, [
            //     str_nested(2, 22), // NAME_WITH_UNDERSCORE
            // ]),
            // but...
            literal(2, 22, [
                reference(2, 7, [  // NAME_
                    reference_target(2, 7, [
                        reference_target_uq(2, 6),
                    ]),
                ]),
                reference(7, 12, [ // WITH_
                    reference_target(7, 12, [
                        reference_target_uq(7, 11),
                    ]),
                ]),
                str_nested(12, 22), // UNDERSCORE
            ]),
        ]
    };
    ...
}
```

Also under `Rule::document`, the parse result has the same output.

```rust
#[test]
fn inline_code_literal_with_underscore() {
    ...

    parses_to! {
        parser: RstParser,
        input: "\
``OK``
``NAME_WITH_UNDERSCORE``
",
        rule: Rule::document,
        tokens: [
            paragraph(0, 32, [
                literal(2, 4, [
                    str_nested(2, 4), // OK
                ]),
                ws_newline(6, 7),
                // I expected this...
                // literal(9, 29, [
                //     str_nested(9, 29), // NAME_WITH_UNDERSCORE
                // ]),
                // but...
                literal(9, 29, [
                    reference(9, 14, [  // NAME_
                        reference_target(9, 14, [
                            reference_target_uq(9, 13),
                        ]),
                    ]),
                    reference(14, 19, [ // WITH_
                        reference_target(14, 19, [
                            reference_target_uq(14, 18),
                        ]),
                    ]),
                    str_nested(19, 29), // UNDERSCORE
                ]),
            ]),
        ]
    };
}
```

It seems that the underscores within double backticks are treated as a part of `reference` the same as the one like "`link`_".
If you could fix this, I would very appreciate it!

I'm learning about the concept of pest parser from your code.
Thank you for this nice library. I love it ;)

**Comment by flying-sheep** [comment]
Thank you for the kind words! I’d hope this library was more complete, but somehow I don’t have the motivation to code much at the moment.

I’ll take a look!

**Comment by flying-sheep** [comment]
Fixed in 6c41888d2b1ee84a5ac7ccded3b15743176289c9

If you want other pieces done, feel free to contribute or hit me up!

**Comment by grauwoelfchen** [comment]
@flying-sheep Thank you for such a quick fix! It worked fine ;)

> If I overlooked something here ory you want other pieces done, feel free to contribute or hit me up!
Much appreciated. I'll try it ᕕ( ᐛ )ᕗ
